### PR TITLE
LibC+LibELF: Implement dladdr()

### PR DESCRIPTION
**Pull request by gunnarbeutner**
**LibC+LibELF: Implement dladdr()**

This implements the `dladdr()` function which lets the caller look up the symbol name, symbol address as well as library name and library base address for an arbitrary address.

**Utilities: Add support for testing null deferencing a RefPtr**

This adds the new flag `-R` for the `crash` utility which tests what happens when we dereference a null `RefPtr`. This is useful for testing the output of the assertion message.

### CODE REVIEW

**Comment by gunnarbeutner** [comment]
One step further would be to use `LibDebug` to get us line information for the symbol address. Unfortunately that would require adding `LibDebug` to `LibC` and I'll leave that for another PR.

File: Userland/Libraries/LibDl/dlfcn.cpp [hunk 1]
```diff
@@ -60,3 +60,15 @@ void* dlsym(void* handle, const char* symbol_name)
...
+{
+    auto result = __dladdr(addr, info);
+    if (result.is_error()) {
+        // FIXME: According to the man page glibc does _not_ make the error
+        // available via dlerror(), however we do. Does this break anything?
+        store_error(result.error().text);
+        return 0;
+    }
+    return 1;
+}
```
Comment (awesomekling) [inline Userland/Libraries/LibDl/dlfcn.cpp:74]: pls configure your editor to put a newline at the end of files :^)

Comment (gunnarbeutner) [reply Userland/Libraries/LibDl/dlfcn.cpp:74]: Every single time. I blame CLion. :)

Comment (IdanHo) [reply Userland/Libraries/LibDl/dlfcn.cpp:74]: `File | Settings | Editor --> Ensure blank line before end of file on Save` in case you havent already :^)

Comment (gunnarbeutner) [reply Userland/Libraries/LibDl/dlfcn.cpp:74]: Neat, let's see if that fixes it in the future. :partying_face:

**Comment by gunnarbeutner** [comment]
I've removed the `assert()` patch because I've found a much better way to do this (our symbols already contain enough information to figure out where a function was inlined).

**Pull request merged**